File: OOPS/inheritance.cpp
#include<iostream>
using namespace std;

class Vehicle{
  public:
    string color = "white";
    void showColor(){
      cout<<"color : "<<color<<endl;
    }
};
class Car : public Vehicle{
  public:
    string model ="sedan";
    void showModel(){
      cout<<"Model : "<<model<<endl;
    }
};
int main(){
  Vehicle v;
  v.showColor(); // color : white
  Car c;
  c.showModel(); // Model : sedan
  return 0;
}

File: OOPS/inheritance.java

class  Vehicle{
  String color = "White";
  public void showColor(){
    System.out.println("color : "+color);
  }
}
class Car extends Vehicle{
  String model = "sedan";
  public void showModel(){
    System.out.println("Model : "+model);
  }
}
public class inheritance {
  public static void main(String[] args) {
    Vehicle v = new Vehicle();
    v.showColor(); // color : White

    Car c = new Car();
    c.showModel(); // Model : sedan
  }
}
